### PR TITLE
feat: Adding JSON schema support to firestore plugin

### PR DESCRIPTION
**Pull request by mnorbury**
fixes: https://github.com/cloudquery/cloudquery/issues/16359

### CODE REVIEW

File: plugins/source/firestore/client/spec.go [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
 	"strings"
 
 	"cloud.google.com/go/firestore"
 )
 
+// Spec is the (nested) spec used by Firestore Source Plugin
 type Spec struct {
-	ProjectID          string `json:"project_id"`
-	UseBase64          bool   `json:"use_base64"`
+	// the ID of the project to use for this client. If not specified, the project id will be auto-detected from the credentials
```
Comment (candiduslynx) [inline plugins/source/firestore/client/spec.go:15]:
```suggestion
	// The ID of the project to use for this client. If not specified, the project id will be auto-detected from the credentials.
```

File: plugins/source/firestore/client/spec.go [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+	ProjectID string `json:"project_id"`
+	// if true the service account JSON content will be treated as base64 encoded
```
Comment (candiduslynx) [inline plugins/source/firestore/client/spec.go:17]:
```suggestion
	// If `true` the `service_account_json` content will be treated as base64-encoded.
```

File: plugins/source/firestore/client/spec.go [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+	UseBase64 bool `json:"use_base64" jsonschema:"default=false"`
+	//  service account JSON content
 	ServiceAccountJSON string `json:"service_account_json"`
-	MaxBatchSize       int    `json:"max_batch_size"`
-	OrderBy            string `json:"order_by"`
-	OrderDirection     string `json:"order_direction"`
+	// maximum batch size for each request when reading Firestore data
```
Comment (candiduslynx) [inline plugins/source/firestore/client/spec.go:21]:
```suggestion
	// Maximum batch size for each request when reading Firestore data.
```

File: plugins/source/firestore/client/spec.go [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
+	MaxBatchSize int `json:"max_batch_size" jsonschema:"minimum=1"`
+	// field(s) to order the results by
```
Comment (candiduslynx) [inline plugins/source/firestore/client/spec.go:23]:
```suggestion
	// List of fields to order the results by.
```

File: plugins/source/firestore/client/spec.go [hunk 5]
```diff
@@ -1,23 +1,34 @@
...
+	OrderBy string `json:"order_by"`
+	// the direction to order the results by when order_by is specified - accepts either asc or desc
```
Comment (candiduslynx) [inline plugins/source/firestore/client/spec.go:25]:
```suggestion
	// The order direction used when `order_by` is `true`.
```

File: plugins/source/firestore/client/spec.go [hunk 6]
```diff
@@ -1,23 +1,34 @@
...
 
+// Spec is the (nested) spec used by Firestore Source Plugin
 type Spec struct {
-	ProjectID          string `json:"project_id"`
-	UseBase64          bool   `json:"use_base64"`
+	// the ID of the project to use for this client. If not specified, the project id will be auto-detected from the credentials
+	ProjectID string `json:"project_id"`
+	// if true the service account JSON content will be treated as base64 encoded
+	UseBase64 bool `json:"use_base64" jsonschema:"default=false"`
+	//  service account JSON content
```
Comment (candiduslynx) [inline plugins/source/firestore/client/spec.go:19]:
```suggestion
	//  Service account JSON content.
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]
I'd like to edit the descriptions/comments, but otherwise good

**Comment by mnorbury** [comment]
> I'd like to edit the descriptions/comments, but otherwise good

👍 Committed the doc changes - and added the workflow to `make gen` that I had missed.

File: .github/workflows/source_firestore.yml [hunk 7]
```diff
@@ -50,6 +50,17 @@ jobs:
           args: "--config ../../.golangci.yml"
           skip-pkg-cache: true
           skip-build-cache: true
+      - name: Setup CloudQuery
+        if: github.event_name == 'pull_request'
+        uses: cloudquery/setup-cloudquery@v3
+        with:
+          version: v5.1.2
```
Comment (candiduslynx) [inline .github/workflows/source_firestore.yml:57]:
```suggestion
```

not required here

**Review by candiduslynx — CHANGES_REQUESTED** [review]
remove unnecessary step from workflow, but otherwise good

File: .github/workflows/source_firestore.yml (hunk 7 above)
Comment (mnorbury) [reply .github/workflows/source_firestore.yml]: Done [here](https://github.com/cloudquery/cloudquery/pull/16380/commits/02d6bdfcdf1ddef7eb10f0fd0ff7ab7d20953ccf)

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**